### DISCUSSION TITLE
How to work with serializers.Serializer?

### DISCUSSION THREAD

**Discussion by SafaAlfulaij** [django-json-api/django-rest-framework-json-api#925] [Q&A] [answered]
How to work with the base `Serializer`?
How to include it? How to give it a pk?
I saw that some code got deprecated at a point of a time: https://github.com/django-json-api/django-rest-framework-json-api/commit/ab3fbfc8020b24c5aab54d53aa08cd927992bd48#diff-52135bfc77d9d40d99d5451d1c64e64d75f0ae5bd338162064891dfcfc27b3b0L114

Currently, it shows as a simple JSON field in the returned data.

**Comment by SafaAlfulaij** [comment, marked as answer]
I think it can just live as a simple attribute, not as an included resource.
That's after a bit of thinking, and checking the folks at JSON:API forums.

**Reply by sliverc** [reply]
You have two ways either you nest it as an attribute simply as nested serializer (possible since version 3.2.0) or you define its own endpoint and connect it with `SerializerMethodResourceRelatedField`.

In the latter case the object it handles need to have a `pk` property (also see #155) defining the primary key.
